### PR TITLE
fix(Core/The Oculus) Drakos dragon keepers

### PR DESCRIPTION
**Pull request by wizzymore**
## CHANGES PROPOSED:
- Make Dragon Keepers outside of the cages
- Fix the broken NPC Texts and gossips, now they are blizzlike
- Add a wait time when npcs leave the cages, as on blizz

## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/3519

## TESTS PERFORMED:
- Kill Drakos
- Wait for them to move out
- Relog, they are still out.
- Tested the chats with different essences
- Restarted the server and did not reset instance, the npcs are in their place.

## HOW TO TEST THE CHANGES:

- Go to Oculus
```
.tele nexus
.gm on
```
- Kill Drakos
- See if the Drakes get out of the cages ( in humanoid form )
- Relog and see if they are still out
- Try to use the gossips of the npcs
- Try to restart the server ( and do not reset instances ) and see if they are still in the correct place
- Try to reset the instance and see if they are in the cages again.

## KNOWN ISSUES AND TODO LIST:

- [x] NPC BELGARISTRASZ is still silent after he finishes his walk
- [x] The gossips are messed up for all of them.
- [x] NPC's don't spawn where they should after server restart and instance in progress ( DRAKOS KILLED )

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by wizzymore** [comment]
" src/server/scripts/Northrend/Nexus/Oculus/oculus.cpp#L150-L215

Complex Method"
Simpler then this, i don't think it can be done, this has gone trough like 5 revisions of code ( only in the last published commits you can see how much the code simplified ) i can't do it more than this tho :D

**Comment by wizzymore** [comment]
This is ready to be tested

**Comment by wizzymore** [comment]
https://www.youtube.com/watch?v=cPbDx1ockIs&feature=youtu.be

How the script worked and how it works now

**Comment by wizzymore** [comment]
Fixed a bug where on server restart the mobs would stay in the cell.

**Comment by wizzymore** [comment]
@Si1ker everything ok now

**Review by Si1ker — APPROVED** [review]

**Comment by Kitzunu** [comment]
https://www.codefactor.io/repository/github/azerothcore/azerothcore-wotlk/pull/3531

**Comment by FrancescoBorzi** [comment]
@azerothcore/testers please have a look at this

**Comment by wizzymore** [comment]
> 
> 
> https://www.codefactor.io/repository/github/azerothcore/azerothcore-wotlk/pull/3531

The majority of them are not fixable.

File: data/sql/updates/pending_db_world/rev_1601843049417097700.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
...are more than a match for the ring guardians, but Varos stands behind an impenetrable shield created from the energy of the Oculus itself.  Ten centrifuge constructs power the shield from the ring and platforms above.  Destroy them and Varos will be vulnerable.$B$BI can grant you the power to call upon a drake from the red flight.  Speak to Eternos or Verdisa if you prefer to draw on the power of the bronze or the green.', '', 0, 0, 0, 0, 0, 0, 0, 0, 1, 18019),(29536, 0, 'Varos Cloudstrider and his ring guardians protect the second ring.  Your drakes are more than a match for the ring guardians, but Varos stands behind an impenetrable shield created from the energy of the Oculus itself.  Ten centrifuge constructs power the shield from the ring and platforms above.  Destroy them and Varos will be vulnerable.', 'Varos Cloudstrider and his ring guardians protect the second ring.  Your drakes are more than a match for the ring guardians, but Varos stands behind an impenetrable shield created from the energy of the Oculus itself.  Ten centrifuge constructs power the shield from the ring and platforms above.  Destroy them and Varos will be vulnerable.', 0, 0, 0, 0, 0, 0, 0, 0, 1, 18019);
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1601843049417097700.sql:24]: Why two blanks after every period?

Comment (wizzymore) [reply data/sql/updates/pending_db_world/rev_1601843049417097700.sql:24]: This is how i found the text, only added breaks where they we re not, i will check how this text look on the npc vs retail and check to see if the spaces are needed or if they change anything.

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1601843049417097700.sql:24]: Cool =)

Comment (wizzymore) [reply data/sql/updates/pending_db_world/rev_1601843049417097700.sql:24]: Seems like it is the same formatting.

**Comment by FrancescoBorzi** [comment]
> > https://www.codefactor.io/repository/github/azerothcore/azerothcore-wotlk/pull/3531
> 
> The majority of them are not fixable.

it's ok, we can ignore that one

**Comment by nicoaravena** [comment]
Tested, working as spected.
NPC are out after server restart
imagen

After reset instances they are inside the cages.
imagen

**Comment by FrancescoBorzi** [comment]
good job @wizzymore and @nicoaravena !

**Comment by wizzymore** [comment]
@nicoaravena thanks for testing

**Comment by Kitzunu** [comment]
https://github.com/azerothcore/azerothcore-wotlk/pull/3531/checks?check_run_id=1687848664

**Comment by Kitzunu** [comment]
Thanks =)

**Pull request merged**